core/src/banking_stage.rs

```diff
@@ -119,7 +119,42 @@ impl BankingStageStats {
         }
     }
 
+    fn is_empty(&self) -> bool {
+        0 == self.process_packets_count.load(Ordering::Relaxed) as u64
+            + self.new_tx_count.load(Ordering::Relaxed) as u64
+            + self.dropped_packet_batches_count.load(Ordering::Relaxed) as u64
+            + self.dropped_packets_count.load(Ordering::Relaxed) as u64
+            + self
+                .dropped_duplicated_packets_count
+                .load(Ordering::Relaxed) as u64
+            + self.newly_buffered_packets_count.load(Ordering::Relaxed) as u64
+            + self.current_buffered_packets_count.load(Ordering::Relaxed) as u64
+            + self
+                .current_buffered_packet_batches_count
+                .load(Ordering::Relaxed) as u64
+            + self.rebuffered_packets_count.load(Ordering::Relaxed) as u64
+            + self.consumed_buffered_packets_count.load(Ordering::Relaxed) as u64
+            + self
+                .consume_buffered_packets_elapsed
+                .load(Ordering::Relaxed)
+            + self.process_packets_elapsed.load(Ordering::Relaxed)
+            + self
+                .handle_retryable_packets_elapsed
+                .load(Ordering::Relaxed)
+            + self.filter_pending_packets_elapsed.load(Ordering::Relaxed)
+            + self.packet_duplicate_check_elapsed.load(Ordering::Relaxed)
+            + self.packet_conversion_elapsed.load(Ordering::Relaxed)
+            + self
+                .unprocessed_packet_conversion_elapsed
+                .load(Ordering::Relaxed)
+            + self.transaction_processing_elapsed.load(Ordering::Relaxed)
+    }
```

Comment:
I think this may be a bit clunky to maintain every time we add a field.

What if we moved all the fields excluding `last_report` and `id` into a separate `BankingStageStatsInner` which derives default, and then this `is_empty()` check can be `self == BankingStageStatsInner::default()`

Reply:
brilliant, would work great if those counter are not `atomic` type. `self.inner == BankingStageStatsInner::default()` would invoke `==` on AtomicU64, which I need to implemented counter by counter, would end to be the same effect. Maybe I missed some neat Rust feature.